ios/FluentUI/Core/Theme/Tokens/TokenResolver.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Contains all possible token overrides for a given control, and returns the appropriate token value for a given property.
+class TokenResolver<ControlType: TokenizedControl>: NSObject, ObservableObject {
+
+    /// Returns the appropriate token value for a given key path on this control's token set.
+    ///
+    /// This method looks for the first value that does not match the default value provided by `defaultTokens`.
+    /// Priority order for tokens are `overrideTokens`, `themeTokens`, then finally `defaultTokens`.
+    func value<ValueType: Equatable>(_ keyPath: KeyPath<TokenType, ValueType>) -> ValueType {
+        let defaultValue = defaultTokens[keyPath: keyPath]
+        if let overrideValue = overrideTokens?[keyPath: keyPath], overrideValue != defaultValue {
+            return overrideValue
+        } else if let themeValue = themeTokens?[keyPath: keyPath], themeValue != defaultValue {
+            return themeValue
+        } else {
+            return defaultValue
+        }
+    }
+
+    /// Prepares all three token sets by installing the current `FluentTheme` and then allowing the control
+    /// to perform additional configuration (e.g. setting `style` or `size` properties) via this class's `tokenConfig` property.
+    func configure(_ tokenConfig: ((_ tokens: TokenType) -> Void)? = nil) {
```

Comment:
Why do we pass in TokenType for additional configuration on the tokens? Would it work with ControlTokens instead for example?

Reply:
> Would it work with ControlTokens instead for example?

Yep! In fact, if you look at line 45, you'll see that `TokenType` is simply an alias to the current control's token type, which itself is required to be of type `ControlTokens`.